dojo/utils.py

```diff
@@ -1505,20 +1509,16 @@ def __init__(self, product_id, title=None, tab=None):
         self.product = Product.objects.get(id=product_id)
         self.title = title
         self.tab = tab
-        self.engagement_count = Engagement.objects.filter(
-            product=self.product, active=True).count()
-        self.open_findings_count = Finding.objects.filter(test__engagement__product=self.product,
+        self.engagement_count = Engagement.objects.filter(product=self.product, active=True).count()
+        self.open_findings_count = Finding.objects.filter(test__engagement__product=product_id,
```

Comment:
Why did you switch to `product_id`, when other selects still use `self.product`?